tests/unit/test_repository.py

```diff
@@ -245,107 +250,245 @@ def test_create_pull_request(repository_client: Client, mock_pull_request: PullR
 
     with pytest.raises(RepositoryClientError) as exc:
-        repository_client.get_file_content(path="path 1")
+        repository_client.tag_commit(tag_name="tag 1", commit_sha="sha 1")
 
     assert_substrings_in_string(("unauthorized", "401"), str(exc.value).lower())
 
 
-def test_get_file_content_unknown_object_error(
+def test_tag_commit_tag_delete_tag_github_error(
     monkeypatch: pytest.MonkeyPatch, repository_client: Client
 ):
     """
-    arrange: given Client with a mocked github repository client that raises an exception
-    act: when get_file_content is called
+    arrange: given tag name and commit sha, Client with a mocked github repository client where a
+        deleting a tag raises GithubException
+    act: when tag_commit is called with the tag name and commit sha
+    assert: RepositoryClientError is raised.
+    """
+    mock_github_repository = mock.MagicMock(spec=Repository)
+    mock_github_repository.get_git_ref.return_value.delete.side_effect = GithubException(
+        status=401, data="Unauthorized", headers=None
+    )
+    monkeypatch.setattr(repository_client, "_github_repo", mock_github_repository)
+
+    with pytest.raises(RepositoryClientError) as exc:
+        repository_client.tag_commit(tag_name="tag 1", commit_sha="sha 1")
+
+    assert_substrings_in_string(("unauthorized", "401"), str(exc.value).lower())
+
+
+def test_tag_commit_tag_not_exists(monkeypatch: pytest.MonkeyPatch, repository_client: Client):
+    """
+    arrange: given tag name and commit sha, Client with a mocked github repository client where
+        retrieving the tag raises UnknownObjectException
+    act: when tag_commit is called with the tag name and commit sha
+    assert: then the functions are called to create the tag.
+    """
+    mock_github_repository = mock.MagicMock(spec=Repository)
+    mock_github_repository.get_git_ref.side_effect = UnknownObjectException(
+        status=404, data="Tag not found error", headers=None
+    )
+    monkeypatch.setattr(repository_client, "_github_repo", mock_github_repository)
+    tag_name = "tag 1"
+    commit_sha = "sha 1"
+
+    repository_client.tag_commit(tag_name=tag_name, commit_sha=commit_sha)
+
+    mock_github_repository.get_git_ref.assert_called_once_with(f"tags/{tag_name}")
+    mock_github_repository.get_git_ref.return_value.delete.assert_not_called()
+
+    mock_github_repository.create_git_tag.assert_called_once_with(
+        tag_name, repository.TAG_MESSAGE, commit_sha, "commit"
+    )
+    mock_github_repository.create_git_ref.assert_called_once_with(
+        f"refs/tags/{tag_name}", mock_github_repository.create_git_tag.return_value.sha
+    )
+
+
+def test_tag_commit_tag_exists(monkeypatch: pytest.MonkeyPatch, repository_client: Client):
+    """
+    arrange: given tag name and commit sha, Client with a mocked github repository client
+    act: when tag_commit is called with the tag name and commit sha
+    assert: then the functions are called to create the tag.
```

Comment:
I'd add `and the function to delete the pre-existing tag is called`

Reply:
good point, should be fixed in next commit